app/scripts/metamask-controller.js

```diff
@@ -832,16 +836,31 @@ export default class MetamaskController extends EventEmitter {
       const keyringOverrides = this.opts.overrides?.keyrings;
 
       const additionalKeyringTypes = [
-        keyringOverrides?.trezor || TrezorKeyring,
         keyringOverrides?.ledger || LedgerBridgeKeyring,
         keyringOverrides?.lattice || LatticeKeyring,
         QRHardwareKeyring,
       ];
 
+      const additionalBridgedKeyringTypes = [
+        {
+          keyring: keyringOverrides?.trezor || TrezorKeyring,
+          bridge: keyringOverrides?.trezorBridge || TrezorConnectBridge,
+        },
+      ];
+
       additionalKeyrings = additionalKeyringTypes.map((keyringType) =>
         keyringBuilderFactory(keyringType),
       );
 
+      additionalBridgedKeyringTypes.forEach((keyringType) =>
+        additionalKeyrings.push(
+          hardwareKeyringBuilderFactory(
+            keyringType.keyring,
+            keyringType.bridge,
+          ),
```

Comment:
To play devil's advocate: I suspect a new factory was added for symmetry with `keyringBuilderFactory`. We also have a separate keyring builder factory for MMI below, it looks like. Adding a separate function also gives us a place to document why we need to have a specific factory for hardware keyrings — why we can't just use `keyringBuilderFactory`.

Or do we think that keyring builder factories in general are inadvisable?

Reply:
@mcmire sorry about the delayed response her. I feel factories are useful in certain situations, such as you mentioned wherein the build type will necessarily change how it is setup, or if there is complicated logic in initializing the keyring. My comment wasn't taking into consideration pre-existing patterns which it probably should have but based solely on the logic represented in the PR it feels like it increases complexity unnecessarily. However if we believe the logic will be extended soon or if it makes sense because there are other keyrings with more complex logic that use factories and by adding a new pattern for keyring initialization we are unnecessarily increasing complexity in a different direction then I would be fine with having a factory here to unify patterns.